File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nset = set(nums)
        ans = 0

        for num in nset:
            if (num-1) not in nset:
                length = 1
                while (num+length) in nset:
                    length += 1
                ans = max(ans, length)

        return ans

File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==1) return nums[0];
        vector<int> dp = nums;
        for(int i=1;i<nums.size();i++){
            dp[i] = max(nums[i], nums[i]+dp[i-1]);
        }
        return *max_element(dp.begin(), dp.end());
    }
};
